apps/studio/components/interfaces/Integrations/CronJobs/CronJobRow.tsx

```diff
@@ -0,0 +1,136 @@
+import { toString as CronToString } from 'cronstrue'
+import { useState } from 'react'
+import { History, Loader2, MoreVertical } from 'lucide-react'
+import Link from 'next/link'
+
+import { useParams } from 'common'
+import { useProjectContext } from 'components/layouts/ProjectLayout/ProjectContext'
+import { CronJob } from 'data/database-cron-jobs/database-cron-jobs-query'
+import { useDatabaseCronJobToggleMutation } from 'data/database-cron-jobs/database-cron-jobs-toggle-mutation'
+import { useSendEventMutation } from 'data/telemetry/send-event-mutation'
+import ConfirmationModal from 'ui-patterns/Dialogs/ConfirmationModal'
+import Table from 'components/to-be-cleaned/Table'
+import {
+  Button,
+  DropdownMenu,
+  DropdownMenuContent,
+  DropdownMenuItem,
+  DropdownMenuSeparator,
+  DropdownMenuTrigger,
+  Label_Shadcn_,
+  Switch,
+} from 'ui'
+import { TelemetryActions } from 'lib/constants/telemetry'
+
+interface CronJobRowProps {
+  job: CronJob
+  onEditCronJob: (job: CronJob) => void
+  onDeleteCronJob: (job: CronJob) => void
+}
+
+export const CronJobRow = ({ job, onEditCronJob, onDeleteCronJob }: CronJobRowProps) => {
+  const { ref } = useParams()
+  const { project: selectedProject } = useProjectContext()
+
+  const [showConfirmModal, setShowConfirmModal] = useState(false)
+
+  const { mutate: sendEvent } = useSendEventMutation()
+  const { mutate: toggleDatabaseCronJob, isLoading } = useDatabaseCronJobToggleMutation()
+
+  // pg_cron can also use "30 seconds" format for schedule. Cronstrue doesn't understand that format so just use the
+  // original schedule when cronstrue throws
+  let schedule = job.schedule
+  try {
+    const scheduleString = CronToString(job.schedule)
+    schedule = scheduleString
+  } catch {}
```

Comment:
I think it'd be cleaner to move this to its own function outside the component but it's clear what it does so no blocking

function safeCronToString(schedule: string) → trycatch → returns string, wdyt?

Reply:
oh yes! good point 🙏